wallet-new/src/Cardano/Wallet/Kernel/DB/Util/IxSet.hs

```diff
@@ -146,6 +148,15 @@ member pk = isJust . view (Lens.at pk)
 size :: IxSet a -> Int
 size = IxSet.size . unwrapIxSet
 
+-- | Safely returns the 'head' of this 'IxSet', but only if it is a singleton
+-- one, i.e. only if it has @exactly@ one element in it. Usually this is
+-- used in tandem with 'getEQ' to witness the existence of exactly one element
+-- in the set indexed by a particular index.
+onlyOne :: IxSet a -> Maybe a
+onlyOne ixset = case IxSet.toList . unwrapIxSet $ ixset of
```

Comment:
We can actually rename this `getOne` as the twin function on the underlying `IxSet`, for consistencies' sake.